### PR TITLE
fix(theme): fix links to catch current locale

### PR DESCRIPTION
**Pull request by bartoszherba**
## Description
Localization routes push and nuxt-links.

## Related Issue
-

## Motivation and Context
Bugfix

## How Has This Been Tested?
Go through 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by Frodigo — DISMISSED** [review]
@bartoszherba looks good to me, thanks!

Can you resolve build issues?

File: packages/theme/pages/Checkout/Shipping.vue [hunk 1]
```diff
@@ -233,7 +233,7 @@
       </div>
       <VsfShippingProvider
         v-if="isShippingDetailsStepCompleted && !dirty"
-        @submit="$router.push('/checkout/billing')"
+        @submit="$router.push(String(localePath('/checkout/billing')))"
```
Comment (bloodf) [inline packages/theme/pages/Checkout/Shipping.vue:236]:
```suggestion
        @submit="$router.push(`${localePath('/checkout/billing')}`)"
```

File: packages/theme/pages/Checkout/UserAccount.vue [hunk 2]
```diff
@@ -233,7 +234,7 @@ export default defineComponent({
       }
 
       if (!hasError.value) {
-        await router.push('/checkout/shipping');
+        await router.push(String(app.localePath('/checkout/shipping')));
```
Comment (bloodf) [inline packages/theme/pages/Checkout/UserAccount.vue:237]:
```suggestion
        await router.push(`${app.localePath('/checkout/shipping')}`);
```

File: packages/theme/pages/MyAccount.vue [hunk 3]
```diff
@@ -111,16 +111,15 @@ export default defineComponent({
     const changeActivePage = async (title) => {
       if (title === 'Log out') {
         await logout();
-        await router.push(localePath({ name: 'home' }));
+        await router.push(String(localePath({ name: 'home' })));
```
Comment (bloodf) [inline packages/theme/pages/MyAccount.vue:114]:
```suggestion
        await router.push(`${localePath({ name: 'home' })}`);
```

File: packages/theme/pages/MyAccount/AddressesDetails.vue [hunk 4]
```diff
@@ -110,19 +112,20 @@ export default defineComponent({
...
-      () => userAddresses.value.filter((address) => address?.id == route.value.query.id).pop()
+      () => userAddresses.value.filter((address) => address?.id == route.value.query.id).pop(),
     );
 
     const isNewAddress = computed(() => !activeAddress.value);
     const editingAddress = computed(() => !!route.value.query.id);
     const changeAddress = async (address) => {
       const addressId = address?.id || 'new';
-      await router.push({ path: '/my-account/addresses-details', query: { id: addressId } });
+      await router.push(String(app.localePath({ path: '/my-account/addresses-details', query: { id: addressId } })));
```
Comment (bloodf) [inline packages/theme/pages/MyAccount/AddressesDetails.vue:128]:
```suggestion
      await router.push(`${app.localePath({ path: '/my-account/addresses-details', query: { id: addressId } })}`);
```

File: packages/theme/pages/MyAccount/MyWishlist.vue [hunk 5]
```diff
@@ -270,10 +272,11 @@ export default defineComponent({
           break;
         case 'BundleProduct':
         case 'ConfigurableProduct':
-          await router.push(`/p/${productGetters.getProductSku(product)}${productGetters.getSlug(
+          const path = `/p/${productGetters.getProductSku(product)}${productGetters.getSlug(
             product,
             product.categories[0],
-          )}`);
+          )}`;
+          await router.push(String(app.localePath(path)));
```
Comment (bloodf) [inline packages/theme/pages/MyAccount/MyWishlist.vue:279]:
```suggestion
          await router.push(`${app.localePath(path)}`);
```

File: packages/theme/pages/Product.vue [hunk 6]
```diff
@@ -449,7 +449,7 @@ export default defineComponent({
       productConfiguration.value.push([label, value]);
 
       await router.push({
-        path: route.value.fullPath,
+        path: String(app.localePath(route.value.fullPath)),
```
Comment (bloodf) [inline packages/theme/pages/Product.vue:452]:
```suggestion
        path: `${app.localePath(route.value.fullPath)}`,
```

File: packages/theme/pages/Shipping.vue [hunk 7]
```diff
@@ -180,7 +180,7 @@
       </div>
       <VsfShippingProvider
         v-if="isFormSubmitted"
-        @submit="$router.push('/checkout/billing')"
+        @submit="$router.push(String(localePath('/checkout/billing')))"
```
Comment (bloodf) [inline packages/theme/pages/Shipping.vue:183]:
```suggestion
        @submit="$router.push(`${localePath('/checkout/billing')}`)"
```

File: packages/theme/components/AppHeader.vue [hunk 8]
```diff
@@ -246,7 +248,7 @@ export default defineComponent({
 
     const handleAccountClick = async () => {
       if (isAuthenticated.value) {
-        await router.push('/my-account');
+        await router.push(String(app.localePath('/my-account')));
```
Comment (bloodf) [inline packages/theme/components/AppHeader.vue:251]:
```suggestion
        await router.push(`${app.localePath('/my-account')}`);
```

File: packages/theme/components/BottomNavigation.vue [hunk 9]
```diff
@@ -71,10 +71,10 @@ export default defineComponent({
     } = useUiState();
     const { isAuthenticated } = useUser();
     const router = useRouter();
-
+    const { app } = useContext();
     const handleAccountClick = async () => {
       if (isAuthenticated.value) {
-        return router.push('/my-account');
+        await router.push(String(app.localePath('/my-account')));
```
Comment (bloodf) [inline packages/theme/components/BottomNavigation.vue:77]:
```suggestion
        await router.push(`${app.localePath('/my-account')}`);
```

**Review by bloodf — CHANGES_REQUESTED** [review]

File: packages/theme/components/CartSidebar.vue [hunk 10]
```diff
@@ -293,7 +295,7 @@ export default defineComponent({
 
     const goToCheckout = async () => {
       const redirectUrl = await initializeCheckout({ baseUrl: '/checkout/user-account' });
-      await router.push(redirectUrl);
+      await router.push(String(app.localePath(redirectUrl)));
```
[the remark this answers is not in the document]
Comment (bartoszherba) [reply packages/theme/components/CartSidebar.vue]: I do not see better elegancy in any of both solutions so I measured the performance and indeed interpolation is about 3 times faster so ye, let's change that ;)

**Review by Frodigo — APPROVED** [review]

**Review by bloodf — APPROVED** [review]

**Pull request merged**